### ISSUE TITLE
README: Installation issues.

### ISSUE THREAD

**Issue by drscotthawley** [GreyGnome/KivyDnD#8]
Thanks for making this.  Wanted to point out that the README.md says to run `sudo python setup.py`

If you do that, you get the error:
```
usage: setup.py [global_opts] cmd1 [cmd1_opts] [cmd2 [cmd2_opts] ...]
   or: setup.py --help [cmd1 cmd2 ...]
   or: setup.py --help-commands
   or: setup.py cmd --help
```

Correct instruction would read `sudo python setup.py install`

**Comment by drscotthawley** [comment]
Also, upon running, one gets an error 
`" ImportError: No module named 'debug_print'"`

Seems that setup.py is not installing the debug_print requirement. This could be fixed by adding
```
install_requires=[
        "debug-print >= 0.6.1",
    ],
```

...in the setup() call in setup.py

**Comment by drscotthawley** [comment]
...although after installing debug_print (`pip install debug_print`), and trying to run any example, they all crash with:

` ImportError: No module named 'debug_print_impl'`
My attempts to track down debug_print_impl have proven fruitless.  Any suggestions?

Thought about trying to bypass your reliance on debug_print, but it's  all through the code.

**Comment by EdvardasDlugauskas** [comment]
Thanks for noticing, @drscotthawley ! A detailed issue is a really big help.

@GreyGnome has updated the code quite a bit while I was mostly inactive, but I'll see what I can do.

**Comment by EdvardasDlugauskas** [comment]
Right now a pull request #9 to merge my branch (which is hopefully working) is open.

What you can do, @drscotthawley if @GreyGnome  doesn't accept in soon enough, is install KivyDnD straight from my branch: 

```pip install -e git+https://github.com/EdvardasDlugauskas/KivyDnD#egg=KivyDnDegg```

It seems to be working for me, try it out and tell me if it works for you, @drscotthawley . 

![image](https://user-images.githubusercontent.com/21359162/35757333-35460c6e-0877-11e8-9e0f-50d50f00e363.png)

**Comment by GreyGnome** [comment]
Thanks. I'll fix/ take a look. I don't know about that message
concerning 'debug_print_impl'.
There should be no such module, but let's see.

On Feb 2, 2018 11:26 AM, "Scott Hawley" <notifications@github.com> wrote:

...although after installing debug_print, and trying to run any example,
they all crash with:

ImportError: No module named 'debug_print_impl'
My attempts to track down debug_print_impl have proven fruitless. Any
suggestions?

—
You are receiving this because you are subscribed to this thread.
Reply to this email directly, view it on GitHub
<https://github.com/GreyGnome/KivyDnD/issues/8#issuecomment-362633828>, or mute
the thread

**Comment by GreyGnome** [comment]
I found my round tuit, and I'm working on these things. That _impl message has me vexed, but I'll take a look. I've been working on some functionality changes and hitherto-unseen bugfixes so give me a little bit. You can rip out debug_print if you find that it's too annoying. I need it right now because... well, I create as many bugs as lines of code, it seems :-( .

**Comment by GreyGnome** [comment]
I don't know why setup would omit debug_print. It should install all modules in the kivydnd directory. After all, there's no difference between it and for example the examples/ directory. 
I feel even less confident about "install_requires". For example, I don't know what this means: "requirements are “Abstract”, i.e. not associated with any particular index..." See https://packaging.python.org/discussions/install-requires-vs-requirements/#id4 . Furthermore, I have no separate version of the debug_print.py file and I'm loathe to maintain one because setup isn't performing as it should.
So for these reasons I have attempted to solve the problem by explicitly listing modules in the py_modules directive. You should be able to find the new setup.py soon.

**Comment by EdvardasDlugauskas** [comment]
@GreyGnome it's not that setup.py doesn't include the `debug_print`, it's the usage that was incorrect. See the changes in [this commit](https://github.com/GreyGnome/KivyDnD/commit/b9fcbede3db164062fc7355dd864cbbc85bd0cf9), `from debug_print import Debug` was changed to `from .debug_print import Debug`, where `.` is a short way of saying "this package" or `kivydnd.`. 

**You merged that commit by the way, the setup.py should be working fine now.**

**Comment by drscotthawley** [comment]
Hey @EdvardasDlugauskas @GreyGnome thanks for looking into this. I'm not against debug_print, I'm just against not being able to try your code!  I could maybe just redefine debug_print...

....But actually maybe your code isn't even what I need: I found your repo because I'm looking for a way to drag-and-drop files from the desktop into Kivy with better control that just Window.bind(on_dropfile=...) -- I want to be able to drop different files into different widgets, rather than just the whole Window.  On closer inspection, your code seems more about dropping widgets onto other widgets (which I'm not even trying to do).   ...Do I have that about right?

**Comment by EdvardasDlugauskas** [comment]
Yeah, you're right, this one is about dragging and dropping widgets.

Thanks for testing the code anyway!

**Comment by GreyGnome** [comment]
Sorry to lead you down the dark and twisty path, @drscotthawley . I'll add a note in the README so folks understand that this can't work outside of Kivy. And, as always, thanks @EdvardasDlugauskas .

**Comment by GreyGnome** [comment]
I'm having a hard time creating a working setup.py file. I'm still exploring this issue. Sorry about that.

**Comment by GreyGnome** [comment]
I think I finally have a working install. Grab the .tar.gz file from the dist directory, and perform: pip install kivydnd-0.5.0.tar.gz

**Comment by GreyGnome** [comment]
I updated the README as well. Check out its current instructions, at your leisure.
